### PR TITLE
Feature/rotating models

### PR DESCRIPTION
**Pull request by David-Hanna**
Includes rotating models in the player selection screen and the rolling bowling ball for kingpin's strike ability as they use the same component.

### CODE REVIEW

File: HeatStroke/GOComponents/Components/ComponentRotateOverTime.cpp [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		ParseNode(p_pBaseNode, fRotationPerSecondX, fRotationPerSecondY, fRotationPerSecondZ);
+	}
+	// Then override with the Override node.
+	if (p_pOverrideNode != nullptr)
+	{
+		ParseNode(p_pBaseNode, fRotationPerSecondX, fRotationPerSecondY, fRotationPerSecondZ);
+	}
+
+	// Change seconds to milliseconds for ease of computation in Update()
+	return new ComponentRotateOverTime(
```
Comment (bradcooper314) [inline HeatStroke/GOComponents/Components/ComponentRotateOverTime.cpp:49]: Is anything changing here or is this a leftover comment?

**Review by bradcooper314 — CHANGES_REQUESTED** [review]
Some kind of desync happens as the player select screen models rotate, probably because the y-axis of the driver isn't vertically aligned with the y-axis of the kart. After 1 or 2 spins, the kart's steering wheel is going into the driver's chest.

The bowling ball looks good though...

File: HeatStroke/GOComponents/Components/ComponentRotateOverTime.cpp (hunk 1 above)
Comment (David-Hanna) [reply HeatStroke/GOComponents/Components/ComponentRotateOverTime.cpp:49]: Leftover.

**Comment by David-Hanna** [comment]
Right, the issue is actually that the center of the game object that it's rotating around isn't the same as the center of the model as created in Blender as a result of "sitting" him in the kart. So this is quite a difficult issue to fix as I've already expanded on the models into animations and would have to change frames upon frames of models to now be centered around the correct point and equally offset to make up for the positional difference.

**Comment by bradcooper314** [comment]
Rather than rotating them individually, can we do something similar to what we do in the races. Have them both be a child of an empty adult GameObject that rotates, and do the usual destroy/create when changing characters except now you would add them again as a child. Would this work?

**Comment by David-Hanna** [comment]
Fixed.

**Review by MatthewWhite — APPROVED** [review]

**Pull request merged**